enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,751 @@
+  that my users do not experience a loss of features or significant down time.
+
+* As an OpenShift administrator, I want to utilize Cluster API as a single point
+  of interaction when using the Cluster Autoscaler for monitoring infrastructure
+  activity so that I can reduce the complexity of my automation and observability tooling.
+
+* As an OpenShift user, I want to focus my attention on a single API for machine
+  management so that I do not have to learn multiple interfaces for the same functionality.
+
+* As an OpenShift developer, I want to ensure that all features of the Cluster
+  Autoscaler continue to work as expected on Cluster API so that our users do
+  not experience a regression of functionality.
+
+* As an OpenShift developer, I want to ensure that the Cluster Autoscaler has
+  a consistent use pattern as we migrate from Machine API to Cluster API so that
+  users are not confused by unexpected behaviors and workflows.
+
+### Goals
+
+* Enable Cluster Autoscaler to use Cluster API resources instead of
+  Machine API resources on OpenShift.
+* Enable Cluster Autoscaler Operator to recognize Cluster API resource
+  targets in addition to Machine API resources.
+* Ensure mirroring of autoscaler specific metadata from Machine API
+  resources to Cluster API resources.
+* Add a conditions slice to the MachineAutoscaler resource status field.
+
+### Non-Goals
+
+* Change the expected namespace of operation for the Cluster Autoscaler or
+  Cluster Autoscaler Operator.
+* Change the expected workflow for ClusterAutoscaler and MachineAutoscaler
+  custom resources.
+* Enable the Cluster Autoscaler to manipulate both Machine API and Cluster
+  API MachineSets at the same time.
+
+## Proposal
+
+Update the Cluster Autoscaler to recognize Cluster API resources instead of Machine API.
+This change would bring our version of the autoscaler closer to the upstream
+version and would allow us to drop some patches we are carrying. The Cluster
+API MachineSet sync controller will be updated to recognize when the
+Cluster Autoscaler has made a change to a Cluster API resource and then sync
+the change to the corresponding Machine API resource, regardless of which resource
+is authoritative. The scope of API changes from the Cluster Autoscaler are limited
+to updating the scale subresource and annotations of MachineSet resources.
+
+Update the Cluster Autoscaler Operator to be namespace aware. This requires
+changing the operator to recognize when it has a Machine API or Cluster API
+reference in a MachineAutoscaler, and then use the appropriate namespace to
+locate the resource. The Cluster API MachineSet sync controller will be updated
+to ensure that when the Cluster Autoscaler Operator adds the autoscaling
+annotations that they are copied to any related resources, regardless of which
+is authoritative.
+
+Update the Cluster API MachineSet sync controller to recognize the
```

Comment:
Perhaps instead the sync controller needs an update to covert the MAPI keyed annotations to CAPI keyed annotations?

That way we wouldn't have the MAPI keys on the CAPI resources